app/lib/missing_permission.rb

```diff
@@ -0,0 +1,9 @@
+class MissingPermission < RuntimeError
+  attr_reader :permission, :provider, :provider_user
+
+  def initialize(permission:, provider:, provider_user:)
```

Comment:
It's against Liskov (and feels a bit weird) to change the constructor of `RuntimeError` like this. I think it would be better to take a message in the first position as normal and not use keyword args (hash args we could splat, maybe?)

Reply:
The intention there is to use a custom error class and require this information, otherwise we don't have enough information for an informative 'access denied' page. But I see your point.

I've changed the error to subclass `StandardError` and accept a hash with the required bits. I've also switched callers to `raise <error instance>, message` syntax.